util/encoding/encoding_test.go

```diff
@@ -643,6 +644,71 @@ func TestEncodeDecodeNull(t *testing.T) {
 	}
 }
 
+func TestEncodeDecodeTime(t *testing.T) {
+	zeroTime := time.Unix(0, 0).UTC()
+
+	// test cases are negative, increasing, duration offsets from the
+	// zeroTime. The positive, increasing, duration offsets are automatically
+	// genarated below.
+	testCases := []string{
+		"-1345600h45m34s234ms",
+		"-600h45m34s234ms",
+		"-590h47m34s234ms",
+		"-310h45m34s234ms",
+		"-310h45m34s233ms",
+		"-25h45m34s234ms",
+		"-23h45m35s",
+		"-23h45m34s999999999ns",
+		"-23h45m34s234ms",
+		"-23h45m34s101ms",
+		"-23h45m34s1ns",
+		"-23h45m34s",
+		"-23h45m33s901ms",
+		"-23h45m",
+		"-23h",
+		"-23612ms",
+		"-345ms",
+		"-1ms",
+		"-201us",
+		"-1us",
+		"-201ns",
+		"-1ns",
+		"0",
+	}
+	// Append all the positive testCases.
+	for i := len(testCases) - 1; i > 0; i-- {
```

Comment:
oh possibly because you're mutating the slice during iteration, though i'm not sure if that makes my suggestion illegal.